### ISSUE TITLE
Clarify that current cross-track distance uses spherical (Haversine) earth model, add Ellipsoidal (Vincenty) version

### ISSUE THREAD

**Issue by urschrei** [georust/geo#1128]
https://docs.rs/geo/latest/geo/algorithm/cross_track_distance/trait.CrossTrackDistance.html

One approach would be:

1. Deprecate `cross_track_distance` in favour of `cross_track_distance_haversine`
2. Add `cross_track_distance_geodesic` (I think we have all the necessary pieces between `geo` and `geographiclib_rs`)
3. Note the advantages / caveats related to WGS-84 (specifically: most WGS-84 coordinates won't be accurate to less than 1 m anyway)

**Comment by frewsxcv** [comment]
A PR was opened for this here: https://github.com/georust/geo/pull/1140

**Comment by frewsxcv** [comment]
(Didn't mean to close)

**Comment by michaelkirk** [comment]
This would be a good candidate to revisit in light of the line measure work I've been doing.

I'd guess the API would look something like:

```
line.cross_track_distance::<Geodesic>(point)
line.cross_track_distance::<Haversine>(point)
```

Also, just to make sure I'm understanding correctly, isn't cross_track_distance **exactly** the same as `Point x Line` distance in a particular metric space?

e.g. we already have implementations:
```
impl Distance<Point, Point> for Haversine { ... }
impl Distance<Point, Point> for Geodesic { ... }
```

If we also had these implementations:
```
impl Distance<Line, Point> for Haversine { ... }
impl Distance<Line, Point> for Geodesic { ... }
```

Than these two are **precisely** the same thing:
```
line.distance::<Haversine>(point);
line.cross_track_distance::<Haversine>(point)
```

Is that right? Please correct me if I'm wrong, I've not worked with "cross track distance" as a concept by name before.

Regardless, even if they _are_ equivalent, it might be worth having it filed under both names since "cross_track_distance" is apparently a term that people know and will try to find.

**Comment by michaelkirk** [comment]
> Than these two are precisely the same thing:
> ```rust
> line.distance::<Haversine>(point);
> line.cross_track_distance::<Haversine>(point)
> ```
> Is that right? Please correct me if I'm wrong, I've not worked with "cross track distance" as a concept by name before.

That is wrong! CrossTrackDistance is measuring the distance to an infinite line passing through the two points. I mistakenly thought it was measuring the distance to the given line segment.
